tools/clang/test/HLSL/globallycoherent-mismatch.hlsl

```diff
@@ -0,0 +1,72 @@
+// RUN: %clang_cc1 -fsyntax-only -ffreestanding -verify %s
+
+RWByteAddressBuffer NonGCBuf;
+globallycoherent RWByteAddressBuffer GCBuf;
+
+RWByteAddressBuffer NonGCBufArr[2];
+globallycoherent RWByteAddressBuffer GCBufArr[2];
+
```

Comment:
Could we have something like this?
static globallycoherent  RWByteAddressBuffer SGCBufArr[2] = NonGCBufArr;
void getNonGCBufArrayArr(inout globallycoherent RWByteAddressBuffer
 TGCBufArr[2]) {
  TGCBufArr = NonGCBufArr;
}